mithril-aggregator/src/tools/single_signature_authenticator.rs

```diff
@@ -0,0 +1,220 @@
+use slog::{debug, Logger};
+use std::sync::Arc;
+
+use mithril_common::entities::{SingleSignatureAuthenticationStatus, SingleSignatures};
+use mithril_common::StdResult;
+
+use crate::MultiSigner;
+
+/// Authenticates single signatures against a signed message.
+pub struct SingleSignatureAuthenticator {
+    multi_signer: Arc<dyn MultiSigner>,
+    logger: Logger,
+}
+
+impl SingleSignatureAuthenticator {
+    /// Creates a new `SingleSignatureAuthenticator`.
+    pub fn new(multi_signer: Arc<dyn MultiSigner>, logger: Logger) -> Self {
+        Self {
+            multi_signer,
+            logger,
+        }
+    }
+
+    /// Authenticates a single signature against a signed message.
+    pub async fn authenticate(
+        &self,
+        single_signature: &mut SingleSignatures,
+        signed_message: &str,
+    ) -> StdResult<()> {
```

Comment:
Why not simply return a boolean as there is no chance that a error is returned here?

Or probably better, we could consume the `SingleSignature` and return a new `SignleSignature`?